libfwupdplugin/fu-device.c

```diff
@@ -5576,11 +5576,8 @@ fu_device_incorporate(FuDevice *self, FuDevice *donor)
 		fu_device_set_backend_id(self, priv_donor->backend_id);
 	if (priv->update_request_id == NULL && priv_donor->update_request_id != NULL)
 		fu_device_set_update_request_id(self, priv_donor->update_request_id);
-	if (!fu_device_has_private_flag(self, FU_DEVICE_PRIVATE_FLAG_REFCOUNTED_PROXY) &&
-	    !fu_device_has_private_flag(donor, FU_DEVICE_PRIVATE_FLAG_REFCOUNTED_PROXY)) {
-		if (priv->proxy == NULL && priv_donor->proxy != NULL)
-			fu_device_set_proxy(self, priv_donor->proxy);
-	}
+	if (priv->proxy == NULL && priv_donor->proxy != NULL)
+		fu_device_set_proxy(self, priv_donor->proxy);
```

Comment:
ohh, that's the wrong way around, no? i.e. we want:

    -       if (!fu_device_has_private_flag(self, FU_DEVICE_PRIVATE_FLAG_REFCOUNTED_PROXY) &&
    -           !fu_device_has_private_flag(donor, FU_DEVICE_PRIVATE_FLAG_REFCOUNTED_PROXY)) {
    +       if (fu_device_has_private_flag(self, FU_DEVICE_PRIVATE_FLAG_REFCOUNTED_PROXY) &&
    +           fu_device_has_private_flag(donor, FU_DEVICE_PRIVATE_FLAG_REFCOUNTED_PROXY)) {

i.e. if **both** are refcounted, copy it across.

Reply:
https://github.com/fwupd/fwupd/pull/7723 does help, removing this revert.

Reply:
Verified pass on physical device up to the latest change [0d4bbe4](https://github.com/fwupd/fwupd/pull/7717/commits/0d4bbe44f787f1714f6770f94565074625fe89df).